src/marqo/tensor_search/models/recency_parameters.py

```diff
@@ -0,0 +1,135 @@
+        }[self]
+
+
+class RecencyParameters(BaseModel):
+    """Parameters for recency-based score modification.
+
+    Allows boosting of documents based on how recent a timestamp field is,
+    with configurable decay functions.
+    """
+
+    recency_field: str = Field(
+        ...,
+        alias="recencyField",
+        description="Name of the timestamp field to use for recency calculation"
+    )
+
+    scale: str = Field(
+        default="7d",
+        description=(
+            "Time scale controlling decay rate. At distance offset+scale, "
+            "the score reaches decay_to value. Format: {number}{unit} where unit is 'd' (days) or 'h' (hours).\n"
+            "Examples: '7d' (7 days), '168h' (168 hours), '0.5d' (12 hours)\n"
+            "- exponential: smooth exponential decay\n"
+            "- linear: constant rate decay\n"
+            "- gaussian: bell curve decay\n"
+            "- binary: step function (no decay until offset+scale, then drops to decay_to)"
+        )
+    )
+
+    offset: str = Field(
+        default="0d",
+        alias="offset",
+        description=(
+            "Grace period before decay begins. Documents within this age receive "
+            "perfect score (1.0) with no decay applied. Decay starts after this period. "
+            "Format: {number}{unit} where unit is 'd' (days) or 'h' (hours).\n"
+            "Examples: '0d' (no grace period), '2d' (2 days), '12h' (12 hours)"
+        )
+    )
+
+    decay_function: DecayFunction = Field(
+        default=DecayFunction.EXPONENTIAL,
+        alias="decayFunction",
+        description="Type of decay function to apply: exponential (smooth decay), linear (constant decay), gaussian (bell curve), binary (step function at threshold)"
+    )
+
+    decay_to: float = Field(
+        default=0.5,
+        gt=0.0,
+        le=1.0,
+        alias="decayTo",
+        description=(
+            "Target score at distance offset+scale, also acts as floor. "
+            "Must be in range (0.0, 1.0]. This is the score a document receives "
+            "at age = offset + scale, and also the minimum score for older documents. "
+        )
+    )
+
+    apply_in_ranking_phase: ApplyInRankingPhase = Field(
+        default=ApplyInRankingPhase.ALL,
+        alias="applyInRankingPhase",
+        description=(
+            "Controls which ranking phases recency scoring is applied in:\n"
+            "- 'all': Apply in all ranking phases (Vespa rank profile and global phase reranking) (default)\n"
+            "- 'only-global': Calculate recency score in Vespa but only apply it during global phase reranking\n"
+            "- 'exclude-global': Apply recency in Vespa rank profile only, exclude from global phase reranking"
+        )
+    )
+
+    class Config:
+        extra: str = "forbid"
+        allow_population_by_field_name = True
+        use_enum_values = True
+
+    @validator('recency_field')
+    def validate_field_name(cls, v: str) -> str:
```

Comment:
If the user sends some weird characters in the field name, will it be forwarded to Vespa and raise a 500 error?

Reply:
Then that's fine.